### PR TITLE
Operation now returns StatusCode Stubs as well

### PR DESCRIPTION
**Pull request by knom**
@mandrean @jimschubert
#7097 

This uses this.StatusCode(..) to render more verbose STUBS for the Controllers.
So people that are less familiar with ASP.NET Core can understand how it works..

### CODE REVIEW

**Comment by jimschubert** [comment]
@knom Thanks this is a really helpful addition.

My only comment is more of a question (or suggestion). Rather than returning actual code, couldn't we just document the returns as todos themselves, for example:

```
// TODO: return StatusCode({{code}});
```

Reason I ask is because many people use static code analysis and `if(false)` is likely to fail compilation while working through these stubs, in which case the developer would have to manually comment out these conditions.

**Comment by jimschubert** [comment]
@knom If you could, it would be faster. I don't have push access to the repo. I really appreciate the contributions!

**Comment by knom** [comment]
@jimschubert done!
Let me know what you thing ;-)

File: modules/swagger-codegen/src/main/resources/aspnetcore/controller.mustache [hunk 1]
```diff
@@ -33,11 +33,18 @@ namespace {{packageName}}.Controllers
         [ValidateModelState]
         [SwaggerOperation("{{operationId}}")]{{#responses}}{{#returnType}}
         [SwaggerResponse({{code}}, typeof({{&returnType}}), "{{message}}")]{{/returnType}}{{/responses}}
-        public virtual {{#returnType}}IActionResult{{/returnType}}{{^returnType}}void{{/returnType}} {{operationId}}({{#allParams}}{{>pathParam}}{{>queryParam}}{{>bodyParam}}{{>formParam}}{{>headerParam}}{{#hasMore}}, {{/hasMore}}{{/allParams}})
-        { {{#returnType}}
+        public virtual IActionResult {{operationId}}({{#allParams}}{{>pathParam}}{{>queryParam}}{{>bodyParam}}{{>formParam}}{{>headerParam}}{{#hasMore}}, {{/hasMore}}{{/allParams}})
+        { {{#responses}}{{#dataType}}
+            //TODO: Uncomment the next line to return response {{code}} or use other options such as return this.NotFound(), return this.BadRequest(..), ...
+            // return StatusCode({{code}}, default({{&dataType}}));
+            {{/dataType}}{{^dataType}}
```
Comment (wing328) [inline modules/swagger-codegen/src/main/resources/aspnetcore/controller.mustache]: @knom Please use the following instead to allow empty lines with spaces:
```
            {{#dataType}}
            ...
            {{/dataType}}
            {{^dataType}}
            ....
            {{/dataType}}
```

**Comment by wing328** [comment]
@knom please run `bin\windows\aspnetcore-petstore-server.bat` to update the petstore samples.

**Comment by knom** [comment]
Just updated the formatting!
Is this what you were looking for @wing328 ?

Your comment wasn't fully clear to me..

THANKS!

File: samples/server/petstore/aspnetcore/src/IO.Swagger/project.json [hunk 2]
```diff
@@ -22,7 +22,8 @@
     "Microsoft.Extensions.Options.ConfigurationExtensions": "1.0.0",
     "Microsoft.EntityFrameworkCore": "1.0.0",
     "Swashbuckle.AspNetCore": "1.0.0-rc3",
-    "Newtonsoft.Json": "9.0.1"
+    "Newtonsoft.Json": "10.0.3",
+    "JsonSubTypes": "1.1.3"
   },
```
Comment (knom) [inline samples/server/petstore/aspnetcore/src/IO.Swagger/project.json]: This is not part of my updates.. note sure why this is changing...

File: samples/server/petstore/aspnetcore/src/IO.Swagger/Models/Pet.cs [hunk 3]
```diff
@@ -68,19 +68,19 @@ public enum StatusEnum
...
-            PendingEnum,
+            PendingEnum = 2,
             
             /// <summary>
             /// Enum SoldEnum for "sold"
             /// </summary>
             [EnumMember(Value = "sold")]
-            SoldEnum
+            SoldEnum = 3
         }
```
Comment (knom) [inline samples/server/petstore/aspnetcore/src/IO.Swagger/Models/Pet.cs]: This is not part of my updates.. note sure why this is changing...

File: samples/server/petstore/aspnetcore/src/IO.Swagger/Models/Order.cs [hunk 4]
```diff
@@ -60,19 +60,19 @@ public enum StatusEnum
             /// Enum PlacedEnum for "placed"
             /// </summary>
             [EnumMember(Value = "placed")]
-            PlacedEnum,
+            PlacedEnum = 1,
```
Comment (knom) [inline samples/server/petstore/aspnetcore/src/IO.Swagger/Models/Order.cs]: This is not part of my updates.. note sure why this is changing...

File: samples/server/petstore/aspnetcore/src/IO.Swagger/Controllers/PetApi.cs [hunk 5]
```diff
@@ -33,31 +33,39 @@ public class PetApiController : Controller
         /// <summary>
         /// Add a new pet to the store
         /// </summary>
-        /// <remarks></remarks>
+
```
Comment (knom) [inline samples/server/petstore/aspnetcore/src/IO.Swagger/Controllers/PetApi.cs]: This is not part of my updates.. note sure why this is changing...

**Comment by wing328** [comment]
> Is this what you were looking for @wing328 ?

Yup, sorry that I wasn't clear.

**Pull request merged**